bench/Main.hs

```diff
@@ -274,14 +372,37 @@ setup = do
   callCommand $ "cabal get -v0 " <> examplePackage <> " -d " <> examplesPath
   writeFile
     (examplesPath </> examplePackage </> "hie.yaml")
-    ("cradle: {cabal: {component: " <> show examplePackageName <> "}}")
+    exampleCradle
 
   whenJust (shakeProfiling ?config) $ createDirectoryIfMissing True
 
   return $ removeDirectoryRecursive examplesPath
 
--- | Asks the server to shutdown and exit politely
-exitServer :: Session ()
-exitServer = request_ Shutdown (Nothing :: Maybe Value) >> sendNotification Exit ExitParams
-
 --------------------------------------------------------------------------------------------
+
+-- Parse the max residency in RTS -s output
```

Comment:
Totally fine to keep this for now but once this starts to become more complex we should switch to the machine-readable output of the RTS stats.

Reply:
Do you mean eventlog? It requires relinking the binary against the eventlog RTS, which would break `cabal bench`.

Reply:
No I mean the machine readable output you get from the standard RTS, see https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/runtime_control.html#rts-options-to-produce-runtime-statistics

Reply:
Oh, that's a good point, thanks for the tip! Definitely worth switching to that now or at some point.